### PR TITLE
fix/describe-target-health: return unavailable target

### PR DESCRIPTION
**Pull request by mtulio**
## Issue:

describe-target-health is not returning the expected structure when the target is new

## Steps to reproduce:

- create the target group (without registering the target)
- send describe-target-health with a target as a parameter

## Current results:

```
botocore.errorfactory.InvalidTargetException: An error occurred (InvalidTarget) when calling the DescribeTargetHealth operation: The specified target does not exist or is not in the same VPC as the target group.
```

## Expected results

```
Valid `FakeHealthStatus`  with reason `Target.NotRegistered`
```

## References

- https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_TargetHealth.html
-

### CODE REVIEW

File: moto/elbv2/models.py [hunk 1]
```diff
@@ -152,7 +152,17 @@ def deregister_terminated_instances(self, instance_ids: List[str]) -> None:
...
         if t is None:
-            raise InvalidTargetError()
+            port = self.port
+            if 'port' in target:
+                port = target["port"]
+            return FakeHealthStatus(
+                target["id"],
+                port,
+                self.healthcheck_port,
+                "unavailable",
```
Comment (mtulio) [inline moto/elbv2/models.py:162]: I am not sure about the status value, I assumed it for the desired `Reason=Target.NotRegistered`

**Review by bblommers — CHANGES_REQUESTED** [review]
Looks like our linter is failing atm.

Would be good to have a test for this as well - I don't think this scenario is covered yet in our tests

**Comment by mtulio** [comment]
> Looks like our linter is failing atm.
> 
> Would be good to have a test for this as well - I don't think this scenario is covered yet in our tests

Hey @bblommers , linter fixed and unit added. LMK WDYT. Thanks

**Review by bblommers — APPROVED** [review]
LGTM - thanks again @mtulio!

**Pull request merged**